views/paymentClass.tpl

```diff
@@ -344,16 +331,48 @@ class WC_Gateway_{key} extends WC_Payment_Gateway {
 			$type           = isset( $_POST['type'] ) ? sanitize_text_field( wp_unslash( $_POST['type'] ) ) : '';
 			$requireCapture = isset( $_POST['require_capture'] ) ? sanitize_text_field( wp_unslash( $_POST['require_capture'] ) ) : '';
 
-			$xmlResponse          = wp_unslash( $_POST['xml'] );
-			$xml                  = new SimpleXMLElement( $xmlResponse );
-			$xmlToJson            = wp_json_encode( $xml->Body->Transactions->Transaction );
-			$jsonToArray          = json_decode( $xmlToJson, true );
+			$order        = new WC_Order( $order_id );
+			$agreement_id = '';
+
+			$xmlResponse = wp_unslash( $_POST['xml'] );
+			$xml         = new SimpleXMLElement( $xmlResponse );
+
+			if ( $type === 'subscriptionAndCharge' || $type === 'subscription' ) {
```

Comment:
What about subscriptionAndReserve? Or that type is not used at all by the system?

Reply:
@emicha we are not using **subscriptionAndReserve** in our plugin at all. As in the documentation, it is mentioned that **Not all acquirers support this type.**. But we can add support and create a configuration for the merchant to choose from **subscription** or **subscriptionAndReserve**.

Reply:
Yes, could you create a ticket for that so we can add it later on?

Reply:
Done